tests/unit/samplegen/test_samplegen.py

```diff
@@ -1048,6 +1317,175 @@ def test_validate_expression_base_attr_is_repeated():
+                        "key": DummyField(),
+                        "value": DummyField(
+                            message=DummyMessage(
+                                fields={
+                                    "mantle": DummyField(
+                                        message=DummyMessage(type="MANTLE_TYPE",
+                                                             fields={}),
+                                    )
+                                },
+                                type="CEPHALOPOD_TYPE"
+                            )
+                        ),
+                    },
+                    type="CEPHALOPODS_TYPE",
+                    options=namedtuple("MessageOptions", ["map_field"])(False)),
+                repeated=True,
+            )
+        },
+        type="MOLLUSC_TYPE"
+    )
+    method = DummyMethod(output=OutputType)
+    v = samplegen.Validator(method)
+    with pytest.raises(samplegen.BadAttributeLookup):
+        v.validate_expression('$resp.cephalopods{"squid"}.mantle')
+
+
+def test_validate_expression_mapped_no_map_field():
+    OutputType = DummyMessage(
+        fields={
+            "cephalopods": DummyField(
+                message=DummyMessage(
+                    fields={
+                        "key": DummyField(),
+                        "value": DummyField(
+                            message=DummyMessage(
+                                fields={
+                                    "mantle": DummyField(
+                                        message=DummyMessage(type="MANTLE_TYPE",
+                                                             fields={}),
+                                    )
+                                },
+                                type="CEPHALOPOD_TYPE"
+                            )
+                        )},
+                    type="CEPHALOPODS_TYPE",
+                    options=namedtuple("MessageOptions", ["map_field"])(False)),
+                repeated=True,
+            )
+        },
+        type="MOLLUSC_TYPE"
+    )
+    method = DummyMethod(output=OutputType)
+    v = samplegen.Validator(method)
+    with pytest.raises(samplegen.BadAttributeLookup):
+        v.validate_expression('$resp.cephalopods{"squid"}.mantle')
+
+
+def test_validate_expression_mapped_no_value():
+    OutputType = DummyMessage(
+        fields={
+            "cephalopods": DummyField(
+                message=DummyMessage(
+                    fields={"key": DummyField()},
+                    type="CEPHALOPODS_TYPE",
+                    options=namedtuple("MessageOptions", ["map_field"])(True)),
+                repeated=True,
+            )
+        },
+        type="MOLLUSC_TYPE"
+    )
+    method = DummyMethod(output=OutputType)
+    v = samplegen.Validator(method)
+    with pytest.raises(samplegen.BadAttributeLookup):
+        v.validate_expression('$resp.cephalopods{"squid"}.mantle')
+
+
+def test_validate_expression_mapped_no_message():
```

Comment:
This is shadowing the previous (repeated) definition....

Reply:
Good catch!